### ISSUE TITLE
Filepond animations do not work when using axios

### ISSUE THREAD

**Issue by yewwoei** [pqina/react-filepond#65]
### Summary
Hi! I'm new to FilePond, and trying to work it into a website while using axios when files are uploaded for greater control. When using axios, the grey progress animation works. However, the red error or green success animations no longer trigger. Am I missing something out to make the animations work with axios?

I have programmed following an alternate method used for VueJs, written by the Filepond creator on StackOverflow [here](https://stackoverflow.com/questions/52220537/how-to-use-filepond-in-vue-js-and-to-upload-files-using-axios).

### How to reproduce

This is my method used for Filepond's process code.

The odd thing is that when I remove the axios component, leaving the regex portion to trigger an error, the red error animation is displayed. With the axios component, I can't seem to trigger an error or success at all. Is there a method or configuration I am missing?

I have looked all over the React and Filepond documentation and experimented with the error() methods to no avail. 

```javascript
  /**
   * Method is designed according to the filepond documentation.
   * 
   * @param {The name of the original file input element.} fieldName 
   * @param {Actual file object to send. Must be wrapped in a `FormData` object.} file 
   * @param {Metadata of the file.} metadata 
   * @param {The FilePond load method.} load 
   * @param {Error method. You can pass in your own message.} error 
   * @param {To update progress bar.} progress 
   * @param {To stop the async method.} abort 
   */
  processFile(fieldName, file, metadata, load, error, progress, abort) {
    // FormData is a Web API that creates a HTML <form> element.
    const formData = new FormData();
    /**
     * Add a key-value pair to the list.
     * 'file' is the name of the field whose data is contained.
     * The value is the file.
     * filename is reported to the server. Default filename is blob.
     */
    formData.append('file', file, file.name);

    // if filename does not match the required format.
    if (!FILE_NAME_REGEX.test(file.name)) {
      console.log("filename does not match!");
      error('Please use filenames in the following format: NAME_FLOOR_ETC');
    }
    
    // related to aborting the request
    const CancelToken = axios.CancelToken
    const source = CancelToken.source();

    // the request itself
    axios({
      headers: getCsrfHeader().headers,
      method: 'post',
      url: 'api/upload_floorplans',
      data: formData,
      cancelToken: source.token,
      onUploadProgress: (e) => {
        // updating progress indicator
        progress(e.lengthComputable, e.loaded, e.total);
      }
    }).then(response => {
      // passing the file id to Filepond
      load(response.data.data.id)
    }).catch((thrown) => {
      if (axios.isCancel(thrown)) {
        console.log('Request cancelled', thrown.message);
      } else {
        // handle error
      }
    });

    // Should expose an abort method so the request can be cancelled
    return {
      // this function is called if the user has tapped the cancel button.
      abort: () => {
        source.cancel('Operation cancelled by the user.');
      }
    }

  }
```

My Filepond configuration
```javascript
<FilePond
          labelFileProcessingError="Invalid name or format"
          allowFileRename={true}
          ref={ref => this.pond = ref}
          files={this.state.files} // files to display
          allowMultiple={true}
          maxFiles={MAX_NUMBER_OF_FILES_PER_UPLOAD}
          maxParallelUploads={MAX_NUMBER_OF_FILES_PER_UPLOAD}
          server={{
            process:(fieldName, file, metadata, load, error, progress, abort) => {
              this.processFile(fieldName, file, metadata, load, error, progress, abort);
            } 
          }}

          oninit={() => this.handleInit() }
          onupdatefiles={(fileItems) => this.handleOnUpdate(fileItems)}
        >
        </FilePond>
```

With the above code, I see this:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/21275222/52893322-f428c200-314f-11e9-9a3a-d9d6f4bb803c.png">
 
### Expected behaviour

<img width="270" alt="image" src="https://user-images.githubusercontent.com/21275222/52893363-4964d380-3150-11e9-8336-826d6071cb90.png">

<img width="315" alt="image" src="https://user-images.githubusercontent.com/21275222/52893446-05260300-3151-11e9-9512-fed8780c7085.png">

### Additional information
The axios code above connects to a Django API. I have tested the file upload and it uploads the file to my server successfully. The animations just do not work for some reason.

| Environment  | Version
|------------- | -----------
| OS           | MacOS 10.14.3
| Device       | MacBook Pro
| Browser      | Google Chrome

PS: Thank you for the library. It's a beautiful interface.

**Comment by rikschennink** [comment]
Thanks for the detailed issue!

It looks like the `load` and `error` callbacks are note called.

Best to check if those points in the code are reached by setting some `console.log` or debug statements.

```js
axios()
  .then(res => {
     // this point is success, select file id from server response in `res`
     load(res)
  })
  .catch(error => {
    // this point is failure, select error message from `error`
    error(error);
  })
```

**Comment by yewwoei** [comment]
Thanks for the help! I checked through my code, turns out that my load was not called as it was constantly throwing an error due to this component:
```javascript
load(response.data.data.id);
```
Turns out that my `response.data.data` would return undefined every time because of how i structured my HTTP responses on the backend. An error would be thrown every time and load will not be called.

Thanks and have a great week ahead!
